feat: add new tokens to the intent feature

0xmilktea opened the pull request.

0xmilktea commented:
- need to update the intent service URL after adding new assets in production is done.

parrot9design commented:
- [ ] Some pairs aren't working, e.g. I can't swap SUI for any of the SUI LSTs in either direction (one field always stays on 0). And if you try to trade more than ~$500 for other pairs, the field you didn't type into either stays on 0 or shows excessive slippage. It's not clear whether the problem is due to intent limits or incomplete UI support.
![Screenshot 2025-01-13 at 10 36 28 AM](https://github.com/user-attachments/assets/e1fa4854-ebda-4321-9482-35490563a70e) ![Screenshot 2025-01-13 at 10 37 14 AM](https://github.com/user-attachments/assets/4cba90c0-04b3-4e1b-9bb0-09c3121c4100)

Let's add a message underneath the `Insufficient liquidity` button for pairs that are over the limit (whatever that is), similar to the Stability Fund limit message:

![Screenshot 2025-01-13 at 11 52 48 AM](https://github.com/user-attachments/assets/90441e61-5d37-443d-88f2-e4365503bc06)

---

- [ ] If you enter an amount in the For field, the Swap field won't autofill. If you enter an amount in the Swap field, then reverse the swap, the Swap field will be empty. It does not do this in the live app.
https://github.com/user-attachments/assets/2b62b442-3cc9-4d67-b4d3-64197da93868


---

- [ ] Fix the message underneath to use the right capitalisation for all SUI LSTs. (Though I don't think this message should even appear as the only option is an intent-based trade?)
![Screenshot 2025-01-13 at 10 36 47 AM](https://github.com/user-attachments/assets/30958c8d-ea8a-44bf-b447-263077996695)

- [ ] The SUI LST capitalisation is also an issue for the `minimum to receive` value. e.g. When there's no amount entered, it uses `vSUI`, but as soon as you enter a value, it changes to `VSUI`. Same with the others.
![Screenshot 2025-01-13 at 10 49 51 AM](https://github.com/user-attachments/assets/b8e1e3ae-195e-40c2-9c15-582e97413c64) ![Screenshot 2025-01-13 at 10 49 56 AM](https://github.com/user-attachments/assets/63018128-89e0-4f89-8276-8cca0823a5d9)
![Screenshot 2025-01-13 at 12 48 45 PM](https://github.com/user-attachments/assets/d9184c6f-96c4-44c0-a503-4477f1ee5af1)

![Screenshot 2025-01-13 at 12 49 46 PM](https://github.com/user-attachments/assets/545756dc-28f6-4fe0-9ccf-9a824fd4e000)

---

- [ ] Update the decimal precision for tBTC
![Screenshot 2025-01-13 at 12 45 42 PM](https://github.com/user-attachments/assets/4af508c1-4c37-4c84-87ea-cde456cc6b55)

This one failed as soon as I signed for it, so do I really need to also cancel it manually? The cancelling request has taken at least a few minutes so far...
![Screenshot 2025-01-13 at 12 45 53 PM](https://github.com/user-attachments/assets/120f3415-4205-49da-9b38-be15d97b4a72)


---

- [ ] Intents only support Sui and Arbitrum currently, right? If so, we need to remove Base as an option for weETH and wstETH swaps.

- [ ] What if you want to swap between one of the new tokens and a non-intent token, e.g. AVAX?

---

- [ ] Whether we do it now or after this PR, we're gonna need to update the Tokens section to support external tokens and make it clear what the limitations are:
![Screenshot 2025-01-13 at 12 42 47 PM](https://github.com/user-attachments/assets/9ab6f3d0-1b12-47d9-b99f-34cb149962db)
![Screenshot 2025-01-13 at 12 41 27 PM](https://github.com/user-attachments/assets/a5e59adc-a489-4442-9802-6599eff78cd6)

0xmilktea commented:
> * [ ]  Some pairs aren't working, e.g. I can't swap SUI for any of the SUI LSTs in either direction (one field always stays on 0). And if you try to trade more than ~$500 for other pairs, the field you didn't type into either stays on 0 or shows excessive slippage. It's not clear whether the problem is due to intent limits or incomplete UI support.
>   ![Screenshot 2025-01-13 at 10 36 28 AM](https://private-user-images.githubusercontent.com/63842577/402388927-e1fa4854-ebda-4321-9482-35490563a70e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NTkwODgsIm5iZiI6MTczNjc1ODc4OCwicGF0aCI6Ii82Mzg0MjU3Ny80MDIzODg5MjctZTFmYTQ4NTQtZWJkYS00MzIxLTk0ODItMzU0OTA1NjNhNzBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDA4NTk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwMDQ5NjJjOTQ2MTRkMGNmNTE4YTQwZDE4OWViODIwMjJmYjA0OTM3MjVkNzRlOTg0NGQ0YTNhZDg5Y2QwNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wur3bMcO-9sQQUKc13FldJOsL4DwgSh2pZ1gLRMSJSU) ![Screenshot 2025-01-13 at 10 37 14 AM](https://private-user-images.githubusercontent.com/63842577/402388929-4cba90c0-04b3-4e1b-9bb0-09c3121c4100.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NTkwODgsIm5iZiI6MTczNjc1ODc4OCwicGF0aCI6Ii82Mzg0MjU3Ny80MDIzODg5MjktNGNiYTkwYzAtMDRiMy00ZTFiLTliYjAtMDljMzEyMWM0MTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDA4NTk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjYTBmZmRjMDMxNWZkNzcwODQ5NjkyYzJkNDcwNGI2M2Q1MDEwNzIxYzA1NDg4MGQ5NjkxMmRmZjFhYjZiNWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ylrsal0p6FY8nqrGlRwdTSbRwcNIpJGMlS-5Qsv0-X0)
> 
> Let's add a message underneath the `Insufficient liquidity` button for pairs that are over the limit (whatever that is), similar to the Stability Fund limit message:
> 
> ![Screenshot 2025-01-13 at 11 52 48 AM](https://private-user-images.githubusercontent.com/63842577/402392706-90441e61-5d37-443d-88f2-e4365503bc06.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NTkwODgsIm5iZiI6MTczNjc1ODc4OCwicGF0aCI6Ii82Mzg0MjU3Ny80MDIzOTI3MDYtOTA0NDFlNjEtNWQzNy00NDNkLTg4ZjItZTQzNjU1MDNiYzA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDA4NTk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmYWJlMGY4ZDJkZmVjZjY1M2U3NThhMTkxYmYzNjhjMzQ0YjYzMjQ3MTFhNTIxMTk4OGQ1OGE0NTE4NDUyMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jO455KqJ9YvkvAwaLEuowERg53hrnHzq1MOiP1dhd-8)
> 
The solver doesn't support swapping between the same chain assets. For example, it doesn't support swapping between SUI and SUI LST. I forwarded your request to the solver developer.
It's due to the intent limit.

0xmilktea commented:
> * [ ]  If you enter an amount in the For field, the Swap field won't autofill. If you enter an amount in the Swap field, then reverse the swap, the Swap field will be empty. It does not do this in the live app.
>   https://github.com/user-attachments/assets/2b62b442-3cc9-4d67-b4d3-64197da93868
The solver doesn't currently support the exact_out swap feature, but the solver developer will implement it soon.

0xmilktea commented:
> * [ ]  Update the decimal precision for tBTC
>   ![Screenshot 2025-01-13 at 12 45 42 PM](https://private-user-images.githubusercontent.com/63842577/402395411-4af508c1-4c37-4c84-87ea-cde456cc6b55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NTkwODgsIm5iZiI6MTczNjc1ODc4OCwicGF0aCI6Ii82Mzg0MjU3Ny80MDIzOTU0MTEtNGFmNTA4YzEtNGMzNy00Yzg0LTg3ZWEtY2RlNDU2Y2M2YjU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDA4NTk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyY2IyNjViOTJlNWYzZjc1OWNmMzYzYTc1N2M1OGFiNmZjNTg1NDgxYjcyNDNhNzIyMWMxYmRhNjY4OWE0ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2_XaL48O9h149NWC9fjoDYqq9Z06JeQhEwB68gTf8Tg)
> 

It's the solver's side issue. the solver developer is currently fixing on it.

0xmilktea commented:
> * [ ]  What if you want to swap between one of the new tokens and a non-intent token, e.g. AVAX?
Currently, it's impossible to swap in one trade. A user can swap a new token for a bridge token like SUI by using the intent feature. and swap SUI for AVAX by using the Balanced pools.

0xmilktea commented:
In my understanding of your last comment, you want to show the solver's limitations and capabilities. solver's limitations and capabilities continue to change because of its limited liquidity and handled trading orders.

for example, the solver had 100 ETH and 10000 SUI. at this time, the solver has the capacity to swap 100ETH. but after the 100ETH trade, the solver has no ETH to support swapping. Ofc, the solver will refill ETH quickly. 
I think the feature will requirement a significant development resources of the solver side.

parrot9design commented:
- [ ] I don't think JitoSOL is supported yet as it's on Solana?

---

> > * [ ]   If you enter an amount in the For field, the Swap field won't autofill. If you enter an amount in the Swap field, then reverse the swap, the Swap field will be empty. It does not do this in the live app.
> >   https://github.com/user-attachments/assets/2b62b442-3cc9-4d67-b4d3-64197da93868
> 
> The solver doesn't currently support the exact_out swap feature, but the solver developer will implement it soon. But it will take a couple of days of work. So I'd like to implement the feature for the next PR.

I'd prefer to tackle this before merging rather than ship with regressions in the UI.

---

> In my understanding of your last comment, you want to show the solver's limitations and capabilities. solver's limitations and capabilities continue to change because of its limited liquidity and handled trading orders.
> 
> for example, the solver had 100 ETH and 10000 SUI. at this time, the solver has the capacity to swap 100ETH. but after the 100ETH trade, the solver has no ETH to support swapping. Ofc, the solver will refill ETH quickly. I think the feature will requirement a significant development resources of the solver side.

For now, let's focus on the max amount the solver has the capacity to swap and ignore the refill part of the equation. Based on what you said here:

> > * [ ]   What if you want to swap between one of the new tokens and a non-intent token, e.g. AVAX?
> 
> Currently, it's impossible to swap in one trade. A user can swap a new token for a bridge token like SUI by using the intent feature. and swap SUI for AVAX by using the Balanced pools.

We need to have something people can reference to know which swaps are supported and which aren't. In the case of swapping e.g. AVAX for afSUI, we can then have a message underneath to say that the pair is incompatible and to reference the external tokens in the next section for supported swaps.

parrot9design commented:
Getting this error immediately after I sign a transaction. 
![Screenshot 2025-01-20 at 9 55 01 AM](https://github.com/user-attachments/assets/a0586d24-62c1-4cb7-882e-4814edb4e7d8)

When I try to cancel it (which is in itself frustrating as the tx failed so I shouldn't need to do anything), it doesn't cancel properly. It changes to `Canceling...` after I sign a tx, but there's no transaction in my wallet history to indicate that I signed anything, and if I go to the Supply or Bridge tab and come back again, it's reverted to `Cancel`.

It's happened too many times now that **I won't do any more testing until this is resolved, as all it's doing is depleting my wallet of funds.**

Also, where are these error messages coming from? They don't match any of the ones Alex shared, and I would like to improve them.

parrot9design commented:
Before we ship this we need to:

- [ ] Remove JitoSOL on Solana from the Swap panel
- [ ] Figure out why these transactions keep failing: https://github.com/balancednetwork/balanced-network-interface/pull/1581#issuecomment-2601021908
- [ ] Change the button text from `Insufficient liquidity` to `Swap not supported` if you set it to an incompatible pair
![Screenshot 2025-01-20 at 5 00 43 PM](https://github.com/user-attachments/assets/eb9de421-213f-405f-ad58-123f6188b479)

- [ ] Check these pairs, which I wasn't able to get a quote against:
- SUI > tBTC 
- afSUI > tBTC
- haSUI > tBTC
- vSUI > tBTC
- tBTC > weETH
- weETH > tBTC
- tBTC > wstETH
- wstETH > tBTC
- afSUI > ETH
- afSUi > weETH
- afSUI > wstETH
- haSUI > ETH
- haSUI > weETH
- haSUI > wstETH
- vSUI > ETH
- vSUI > weETH
- vSUI > wstETH
- weETH > wstETH
- wstETH > weETH

I've opened a new issue to track Balanced vs external tokens in the Tokens section: https://github.com/balancednetwork/balanced-network-interface/issues/1598

alexandre-chirouze commented:
> Getting this error immediately after I sign a transaction. ![Screenshot 2025-01-20 at 9 55 01 AM](https://private-user-images.githubusercontent.com/63842577/404694531-a0586d24-62c1-4cb7-882e-4814edb4e7d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjAzMDIsIm5iZiI6MTczNzM2MDAwMiwicGF0aCI6Ii82Mzg0MjU3Ny80MDQ2OTQ1MzEtYTA1ODZkMjQtNjJjMS00Y2I3LTg4MmUtNDgxNGVkYjRlN2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIwVDA4MDAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmMTlkOTQ3ZTA1YmQ5MDRjZDU2OTRhZGVjODQxMThiNzA0MmViOWJhMjVmMTE4YmUwNTdhOTI2Y2NhMTY5ZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PtsyreIVv_fJR4owyvjh-GHtD6USmRpxryYLIRpQ3do)
> 
> When I try to cancel it (which is in itself frustrating as the tx failed so I shouldn't need to do anything), it doesn't cancel properly. It changes to `Canceling...` after I sign a tx, but there's no transaction in my wallet history to indicate that I signed anything, and if I go to the Supply or Bridge tab and come back again, it's reverted to `Cancel`.
> 
> It's happened too many times now that **I won't do any more testing until this is resolved, as all it's doing is depleting my wallet of funds.**
> 
> Also, where are these error messages coming from? They don't match any of the ones Alex shared, and I would like to improve them.

@parrot9design This issue is resolved now. These error messages are part of the code I've sent you. When the solver sends an error, it sends an error code (the ones that I sent you on Discord) + an explanation message or data. Right now, the front-end only displays the error message as sent by the solver, but it's up to you to write your own custom messages if you want !


> Before we ship this we need to:
> 
> * [x]  Remove JitoSOL on Solana from the Swap panel
> * [ ]  Figure out why these transactions keep failing: [feat: add new tokens to the intent feature #1581 (comment)](https://github.com/balancednetwork/balanced-network-interface/pull/1581#issuecomment-2601021908)
> * [x]  Change the button text from `Insufficient liquidity` to `Swap not supported` if you set it to an incompatible pair
>   ![Screenshot 2025-01-20 at 5 00 43 PM](https://private-user-images.githubusercontent.com/63842577/404735988-eb9de421-213f-405f-ad58-123f6188b479.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjQwMDksIm5iZiI6MTczNzM2MzcwOSwicGF0aCI6Ii82Mzg0MjU3Ny80MDQ3MzU5ODgtZWI5ZGU0MjEtMjEzZi00MDVmLWFkNTgtMTIzZjYxODhiNDc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIwVDA5MDE0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5NzllZGE5ZjkyNWExNjE0N2UxNDAwMWJlMDJmZDQwMDQ4MzM2Y2I1ZGZjMjA2ZWQxZDdmMzllZjQ3OTU4MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8dhsMOOV7b9A76SjpRD34xRUjO_pyKZAqabVAJl0CdA)
> * [ ]  Check these pairs, which I wasn't able to get a quote against:
> * SUI > tBTC
> * afSUI > tBTC
> * haSUI > tBTC
> * vSUI > tBTC
> * tBTC > weETH
> * weETH > tBTC
> * tBTC > wstETH
> * wstETH > tBTC
> * afSUI > ETH
> * afSUi > weETH
> * afSUI > wstETH
> * haSUI > ETH
> * haSUI > weETH
> * haSUI > wstETH
> * vSUI > ETH
> * vSUI > weETH
> * vSUI > wstETH
> * weETH > wstETH
> * wstETH > weETH
> 
> I've opened a new issue to track Balanced vs external tokens in the Tokens section: #1598

The xCall USDC Bridge between Arbitrum and SUI was empty, which caused the solver unable to find any path between any tokens. This solved by now and an alert will be added for this not to happen in the future.

Those pairs are currently working
SUI > tBTC => OK
afSUI > tBTC => OK
haSUI > tBTC => OK
vSUI > tBTC => OK
afSUI > ETH => OK
afSUi > weETH => OK
afSUI > wstETH => OK
haSUI > ETH => OK
haSUI > weETH => OK
haSUI > wstETH => OK
vSUI > ETH => OK
vSUI > weETH => OK
vSUI > wstETH => OK


Those pairs are current NOT working, I'm debuging this right now and let you know when it's solved
tBTC > weETH => NOT OK
weETH > tBTC => NOT OK
tBTC > wstETH => NOT OK
wstETH > tBTC => NOT OK
weETH > wstETH => NOT OK
wstETH > weETH => NOT OK

alexandre-chirouze commented:
All the paths are now OK !

tBTC > weETH => OK
weETH > tBTC => OK
tBTC > wstETH => OK
wstETH > tBTC => OK
weETH > wstETH => OK
wstETH > weETH => OK

parrot9design commented:
Thanks @alexandre-chirouze!

Still reworking the error messages, should I share them with you or @0xmilktea when they're ready?

I got this one a couple of times, which error code does that map to?
![Screenshot 2025-01-21 at 3 46 45 PM](https://github.com/user-attachments/assets/75f05333-d9c1-4aad-9532-ada8e276e611)



---

- [ ] Fix the decimal precision for the main value in tBTC transactions:
![Screenshot 2025-01-21 at 3 21 40 PM](https://github.com/user-attachments/assets/1757ba0b-6666-4fa1-b251-2e46767a30fb)

---

- [ ] Improve the `Swap not supported` handling.
- This message appears whenever only one field is filled, but it should use `Enter amount` or `Enter {asset 1} amount` here instead:
![Screenshot 2025-01-21 at 3 25 53 PM](https://github.com/user-attachments/assets/1572a7b7-af79-40db-bcbe-d464d4f814bf)

- `Swap not supported` should only appear if both assets are not supported via intents. It should override other messages, like this one:
![Screenshot 2025-01-21 at 3 59 20 PM](https://github.com/user-attachments/assets/ebfefd82-e2bb-4785-b5d6-921cb9a64ca9)

And `Swap not supported` should appear before you enter an amount, as we know which assets are/aren't supported:
![Screenshot 2025-01-21 at 4 26 08 PM](https://github.com/user-attachments/assets/47d68e4c-4127-4993-b834-e450b21e7248)

---

- [ ] My testing started out well, but now the For assets never appear in my wallet. Reproduced with afSUI, haSUI, vSUI, and weETH so far, so I've stopped testing for now to avoid burning through all the crypto in my test wallet. Raised in Discord also.

alexandre-chirouze commented:
@parrot9design 
The Transaction Creation Failed is not part of Solver error message, so the error is from somewhere else before (Maybe the SDK ?)

Solver has a problem right now. For every intent you created and failed, the solver told the front-end that it failed, so the front-end should have asked you to cancel your Intent, was it the case ?

alexandre-chirouze commented:
@parrot9design The bug is now solved. Thanks for finding this, it was not an easy debug and saved us from production nightmare. It also showed that my non-regression tests are not sufficient because it did not catch the bug you found. 
You can start testing again now, sorry for the delay.

@0xmilktea The front-end did not show @parrot9design intents were failed and did not propose to cancel them. From the discussion we had last time on Discord, I've changed the intent status codes, can you check if they are matching ?

Current status are : 
`  NOT_FOUND = -1 // Should happen only if /status is called before /execution endpoint or giving bad task_id
    NOT_STARTED_YET = 1 // Wait...
    STARTED_NOT_FINISHED = 2 // Wait...
    SOLVED = 3 // Worked !
    FAILED = 4 // Fail, tell the user to cancel `

0xmilktea commented:
> @parrot9design The bug is now solved. Thanks for finding this, it was not an easy debug and saved us from production nightmare. It also showed that my non-regression tests are not sufficient because it did not catch the bug you found. You can start testing again now, sorry for the delay.
> 
> @0xmilktea The front-end did not show @parrot9design intents were failed and did not propose to cancel them. From the discussion we had last time on Discord, I've changed the intent status codes, can you check if they are matching ?
> 
> Current status are :
> 
> ```
>     NOT_STARTED_YET = 1 // Wait...
>     STARTED_NOT_FINISHED = 2 // Wait...
>     SOLVED = 3 // Worked !
>     FAILED = 4 // Fail, tell the user to cancel ```
> ```

I just updated it. I forgot to update that. @alexandre-chirouze

alexandre-chirouze commented:
One last thing @0xmilktea, I cannot use my SUI Wallet on the vercel app, is it intended ? 
When clicking on "Echanger", nothing happens and get that error in the console. 
<img width="1720" alt="image" src="https://github.com/user-attachments/assets/655884ab-2163-4258-9278-f36cbe44c8c3" />

0xmilktea commented:
> One last thing @0xmilktea, I cannot use my SUI Wallet on the vercel app, is it intended ? When clicking on "Echanger", nothing happens and get that error in the console. Update : Same with Hana actually, I don't know if you can reproduce ?
> 
> <img alt="image" width="1720" src="https://private-user-images.githubusercontent.com/105735315/405299623-655884ab-2163-4258-9278-f36cbe44c8c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0Nzc2NjUsIm5iZiI6MTczNzQ3NzM2NSwicGF0aCI6Ii8xMDU3MzUzMTUvNDA1Mjk5NjIzLTY1NTg4NGFiLTIxNjMtNDI1OC05Mjc4LWYzNmNiZTQ0YzhjMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyMVQxNjM2MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWU4OWJjYjY0NThjYzExOTA1M2YyZGFmMWZkYmJjZTkwMDU3YzNjN2Y5YTc4Zjc2N2MxMDBmMjJhYWM4MmE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ydYkMseU3H-gsDgYw1Vw7_hdKu7jZY6WOn3UzHeJnj0">

it works on my side.

parrot9design commented:
Mentioned in Discord, but I'll share here, as well.

I’ve tested (almost) all of the 56 different swap variations.

- All swaps from Arb > Sui hung until I cancelled them, but Sui > Sui and Arb > Arb swaps went through without any issue.
- Most Sui > Arb swaps also went through, but vSUI > tBTC, haSUI > ETH, and weETH > haSUI hung and would not process the order cancellation.

I haven’t tested ETH > tBTC as I didn’t have enough to quote, nor have I tested haSUI > wstETH, vSUI > weETH, or ETH <> SUI, because I refreshed the page and now it always white screens.

---

>The Transaction Creation Failed is not part of Solver error message, so the error is from somewhere else before (Maybe the SDK ?)

@0xmilktea or @R0bi7 do you know where I can find error messages used in the SDK?

alexandre-chirouze commented:
@parrot9design It's solved. I'll be working today on my own tests because they definitely fails to catch most of the errors and you shouldn't find a crash everytime you test something. All the bugs you see since 2-3 days are mostly due to same-chain swaps that I have poorly integrated.

I have automated tests that are doing swaps, but they were only checking if the user receives the funds on main pairs, so it wasn't able to catch problems with some other pairs and wasn't actually looking at if the solver received the funds.

Problem is solved but I'm still working on my new tests for that not to happen again.

alexandre-chirouze commented:
@parrot9design I've been working this morning to improve automated tests on the solver to avoid crashes everytime you test anything new.
We are now doing way more pairs swaps to test, and the automated tests check for intent solving and full rebalancing.

Here are the results on the current staging server : 

1 - Arbitrum to SUI
<img width="624" alt="image" src="https://github.com/user-attachments/assets/57495bdf-b290-4857-8570-f7667eed102a" />

2 - SUI to Arbitrum
<img width="595" alt="image" src="https://github.com/user-attachments/assets/cc4d38a3-b7aa-4906-ba9c-c9da37767262" />

3 - Arbitrum to Arbitrum
<img width="605" alt="image" src="https://github.com/user-attachments/assets/cd40f6a4-3441-457f-b39f-84deda2dc7fb" />

4 - SUI to SUI
<img width="588" alt="image" src="https://github.com/user-attachments/assets/0d410470-5f08-497c-993c-8d49d95b5981" />

So this time you should be able to actually test !!

parrot9design commented:
- [ ] These trades from yesterday still won't cancel. Also, can we format them within a single `<p>` tag to avoid dealing with weird columns like the second trade?
![Screenshot 2025-01-23 at 9 36 41 AM](https://github.com/user-attachments/assets/4e44469f-981f-451e-a9d9-8b13059acb3e)


- [ ] Is there any way to detect that pairs like this aren't supported before you enter a value?
![Screenshot 2025-01-23 at 10 04 27 AM](https://github.com/user-attachments/assets/dc31a615-9e96-42fc-9603-8aa5c573fd97) ![Screenshot 2025-01-23 at 10 04 34 AM](https://github.com/user-attachments/assets/6a78f545-1fab-4efc-add6-f519f444b624)

- [ ] I checked the collateral as requested, and noticed the list is much shorter in this build versus the live app:
<img width="350" src="https://github.com/user-attachments/assets/4aeb4489-76b6-48d7-9d19-af55a940c271"> <img width="350" src="https://github.com/user-attachments/assets/2bb44f3c-b94c-4e7b-9fe2-7a3b60a7cabf">

If I try to deposit afSUI, or use the `All collateral` tab to choose another asset I hold, like haSUI, it doesn't let me preview the amount of bnUSD I'll be able to borrow:
![Screenshot 2025-01-23 at 10 23 22 AM](https://github.com/user-attachments/assets/94322b2b-8bca-4cf0-90a9-9c0c6c9a0d60)

This is also an issue in the live app but I've only just noticed: AVAX and BNB prompt me to add some to my wallet even though I hold a small amount of each, but AVAX worked when I increased my balance. I assume the threshold for these tokens has been set higher than some of the others? Coins with a lower balance still allowed me to deposit.
![Screenshot 2025-01-23 at 10 23 44 AM](https://github.com/user-attachments/assets/2a3aae1a-6273-4527-8dca-2b91bd43e9c4)
